src/App.js

```diff
@@ -202,10 +207,19 @@ class App extends React.Component {
         this.setState({ wrapper })
       })
       .catch(err => {
-        console.error('Wrapper init error:', err)
+        log(`Wrapper init error: ${err.name}. ${err.message}.`)
+        this.setState({ appsStatus: APPS_STATUS_ERROR })
       })
   }
 
+  handleRequestAppsReload = () => {
+    this.setState({ appsStatus: APPS_STATUS_LOADING, apps: [] })
+    clearTimeout(this._requestAppsTimer)
+    this._requestAppsTimer = setTimeout(() => {
+      this.updateDao(this.state.locator.dao)
+    }, 1000)
```

Comment:
Is the delay mostly a fudge factor for throttling / network issues / etc.?

Reply:
It is so that if the connection fails immediately, we have the time to see “Loading” (as you can see in the capture), instead of jumping back to the error state immediately.